webapp/tests/karma/ts/services/transitions/create-user-for-contacts.transition.spec.ts

```diff
@@ -0,0 +1,528 @@
+      });
+
+      it(`throws an error if an error is encountered getting the new contact`, async () => {
+        userContactService.get.resolves(ORIGINAL_CONTACT);
+        medicDb.get.rejects({ message: 'Server Error' });
+
+        try {
+          await transition.run([REPLACE_USER_DOC]);
+          assert.fail('should have thrown an error');
+        } catch (err) {
+          assert.equal(err.message, `Server Error`);
+        }
+
+        assert.equal(userContactService.get.callCount, 1);
+        assert.equal(dbService.get.callCount, 1);
+        assert.equal(medicDb.get.callCount, 1);
+        assert.deepEqual(medicDb.get.args[0], [NEW_CONTACT._id]);
+        assert.equal(createUserForContactsService.setReplaced.callCount, 0);
+        assert.equal(createUserForContactsService.isReplaced.callCount, 2);
+      });
+
+      it('throws an error if multiple replace user reports are submitted', async () => {
+        const originalUser = { ...ORIGINAL_CONTACT };
+        userContactService.get.resolves(originalUser);
+        const parentPlace = { ...PARENT_PLACE };
+        medicDb.get
+          .withArgs(PARENT_PLACE._id)
+          .resolves(parentPlace);
+
+        try {
+          await transition.run([REPLACE_USER_DOC, NEW_CONTACT, REPLACE_USER_DOC]);
+          assert.fail('should have thrown an error');
+        } catch (err) {
+          assert.equal(err.message, `Only one user replace form is allowed to be submitted per transaction.`);
+        }
+
+        assert.equal(userContactService.get.callCount, 1);
+        assert.equal(dbService.get.callCount, 0);
+        assert.equal(medicDb.get.callCount, 0);
+        assert.equal(createUserForContactsService.setReplaced.callCount, 0);
+        assert.equal(createUserForContactsService.isReplaced.callCount, 1);
+      });
+    });
+
+    describe(`when the reports submitted do not include a replace user report, but the user is replaced`, () => {
+      afterEach(() => {
+        // Functions from the user replace flow should not be called
+        assert.equal(dbService.get.callCount, 0);
+        assert.equal(medicDb.get.callCount, 0);
+        assert.equal(createUserForContactsService.setReplaced.callCount, 0);
+      });
+
+      it('re-parents the given reports to the new contact', async () => {
+        userContactService.get.resolves(ORIGINAL_CONTACT);
+        createUserForContactsService.isReplaced.returns(true);
+        createUserForContactsService.getReplacedBy.returns(NEW_CONTACT._id);
+        const submittedDocs: any = [
+          getDataRecord(),
+          undefined,
+          deepFreeze(getDataRecord({ _id: undefined })),
+          deepFreeze(getDataRecord({ _id: 'different_contact' })),
+          getDataRecord(),
+          NEW_CONTACT,
+          null
+        ];
+
+        const docs = await transition.run(submittedDocs);
+        assert.deepEqual(docs, [submittedDocs[0], ...submittedDocs.slice(2, 6)]);
```

Comment:
Running assertions by accessing array elements by index is fragile, because the array could get sorted by the code.  Please either freeze the array (Object.freeze will prevent sorting) or define array elements separately, and run assertions over those. 
From a readability perspective, `submittedDocs[4]` is going to be difficult to read.